src/yarnrcYml.ts

```diff
@@ -0,0 +1,25 @@
+import * as path from 'path';
+import { fileSystemOperations } from './fileSystemOperations';
+
+function getYarnrcYmlContents(): Promise<unknown> {
+  return fileSystemOperations.readYaml(path.resolve('.yarnrc.yml'));
+}
+
+export function doesYarnrcYmlFileExist(): Promise<boolean> {
+  const absolutePathToDestinationProjectYarnrcYml = path.resolve('.yarnrc.yml');
+  return fileSystemOperations.pathExists(absolutePathToDestinationProjectYarnrcYml);
+}
+
+export async function isYarnBerryUsingNodeModulesLinker(hasYarnrcYml: Promise<boolean>): Promise<boolean> {
+  if (!(await hasYarnrcYml)) {
+    return false;
+  }
+
+  try {
+    const yarnrc = (await getYarnrcYmlContents()) as { nodeLinker: string };
```

Comment:
I am a little surprised TS is allowing you to cast the `unknown` type in this way. I wonder if we are missing a compiler flag 🤔 
My guess is we would prefer to check if this is an object and then that `nodeLinker` is not undefined and is a string. Try/catching would work but not my preference for an expected scenario.

Reply:
The try/catch is there not for dereferencing `yarnrc`, but in case anything goes wrong with accessing or parsing the file.

Reply:
Do you want me to declare a YarnrcYml interface to encapsulate `nodeLinker`?

Reply:
make it optional, add interface

Reply:
done. I added comments to make it clear what the try/catch is for.